### ISSUE TITLE
[Feature request] Swap steam parser with ValveKeyValue

### ISSUE THREAD

**Issue by beeradmoore** [beeradmoore/dlss-swapper#78]
**Is your feature request related to a problem? Please describe.**
The current parsing of steam libraries is manual and jank. We should use a system that is built specifically to handle it.

**Describe the solution you'd like**
Implement [ValveKeyValue](https://github.com/SteamDatabase/ValveKeyValue/)

**Additional context**

**Comment by MorGuux** [comment]
In my fork (that I admittedly made a while ago and haven't touched since, :( ), I used a different steam library search method (using VDF.NET IIRC) [here](https://github.com/MorGuux/dlss-swapper/blob/main/DLSS%20Swapper/DLSS%20Swapper/Data/SteamLibrary.cs#L59).

I should've continued to work on this fork, but the only thing stopping me from finishing it and PR'ing it was the lack of support for game library grouping (since I used an ObservableCollection with binding so that games added asynchronously would appear as they are found). Maybe we can work something out with that, otherwise my DLSS .DLL searcher function [here](https://github.com/MorGuux/dlss-swapper/blob/main/DLSS%20Swapper/DLSS%20Swapper/DLSSDLLSearcher.cs) is much faster at detecting the DLLS (since it scans each HDD/SSD simultaneously).

**Comment by beeradmoore** [comment]
With the update to v1.1 a lot of code has moved around so I was thinking more about this. I was thinking of making a threaded based system so when a game is detected it is thrown on a queue depending on the drive.

C: -> gameX, gameY, gameZ
E: -> gameA, gameB, gameC, gameD

And then either have a single thread worker assigned per drive (or multiple, or dynamic based on if its a SSD or HDD) and then they can churn through the tasks.

Currently it is doing it per library, and possibly doing all libraries at once.

**Comment by MorGuux** [comment]
Absolutely. I had a system in place that could display all games upon startup before the dll checks began, then simultaneously searched each system disk for its own list of game folders, and updated the displayed game's dll version when found.

Definitely made the UX nicer having all games be displayed at the start, rather than after all folders had been searched.

I'm not familiar with the updated code, but I'd happily take a look at reimplementing the feature in v1.1.

**Comment by beeradmoore** [comment]
I think v1.1 adds games as they are loaded and keeps and activity indicator on them in the DLLs are still being detected.

On top of that it also caches all the games in its own DB, so the games are instantly there. It also caches the last DLLs that were there so rather than scan every game every time it just looks at the files and if they are the same version it expects then it considers it done.

So even without optimised per disk loading it should be improved

**Comment by MorGuux** [comment]
It might be worth fetching file paths from the MFT if the drive uses an NTFS file system. A lib like [NTFSDirect](https://github.com/rasberry/NTFSDirect) could do it (might need some modifications to start looking in a given directory, rather than searching the whole disk.

If the filesystem doesn't have an MFT, then revert to the existing implementation.

**Comment by beeradmoore** [comment]
Yeah that was also another option mentioned in #143

If we are loading cached games and known DLL paths from we wouldn't have to do the disk scan as we already know the paths of what we are looking for. But it could be useful for initial scan + when people hit reload.
